app/server/lib/AttachmentFileManager.ts

```diff
@@ -427,15 +445,17 @@ export class AttachmentFileManager {
   // Uploads the file to an attachment store, overwriting the current DB record for the file if successful.
   private async _storeFileInAttachmentStore(
     store: IAttachmentStore, fileIdent: string, fileData: Buffer
-  ): Promise<string> {
+  ): Promise<void> {
+    if (this._loopAbort.aborted) {
+      throw new Error("AttachmentFileManager was shut down");
+    }
```

Comment:
What's the thinking behind specifically erroring in this method, and not others? (E.g _storeFileInLocalStorage).

Reply:
Look at this again, I'd guess the internal storage would have been the issue due to the DocStorage being shutdown and closing the DB connection. I think remote uploads would be fine, as the stores aren't tied to ActiveDoc for their connections .

I think you've guarded against the internal storage being a problem with the changes to ActiveDoc (as ActiveDoc awaits shutdown for this before closing DocStorage).

I'd be inclined to remove this - I'm not sure it's actually protecting against anything now, and might cause more errors (as I think the transfer would succeed without this guard in place).